tasks/perun_openldap_config.yml

```diff
@@ -132,51 +132,95 @@
   when: ansible_facts.services['slapd'] is not defined
   service_facts:
 
-- name: "Add limits to /etc/security/limits.conf"
-  lineinfile:
-        path: /etc/security/limits.conf
-        state: present
-        line: "{{ item }}"
-  loop:
-    - 'openldap soft nofile 20480'
-    - 'openldap hard nofile 40960'
-  notify: Reload limits
-
-- name: "Add system limit to /etc/sysctl.conf"
-  lineinfile:
-        path: /etc/sysctl.conf
-        state: present
-        line: 'fs.file-max = 500000'
-  notify: Reload sysctl
-
-- name: "Create systemd override config directory if it does not exist"
-  file:
-        path: /etc/systemd/system/slapd.service.d
-        state: directory
-        mode: '0755'
-
-- name: "Add LimitNOFILE to /etc/systemd/system/slapd.service.d/override.conf"
-  blockinfile:
-        path: /etc/systemd/system/slapd.service.d/override.conf
-        create: yes
-        block: |
-          [Service]
-          LimitNOFILE=40960
-  notify: Reload systemd
-
-- name: "Check open file limit using ulimit"
-  command: bash -c 'ulimit -n'
-  register: ulimit_result
-  changed_when: false
-
-- name: "Check system open file limit using sysctl"
-  command: sysctl fs.file-max
-  register: sysctl_result
-  changed_when: false
-
-- name: "Display results of limit checks"
-  debug:
-        msg:
-          - "Current open file limit (ulimit -n): {{ ulimit_result.stdout }}"
-          - "System max open files limit (fs.file-max): {{ sysctl_result.stdout }}"
 
+- name: "set up limits for max open files for LS instance and display it"
+  when: perun_ldap_max_open_files_limits
+  block:
+    - name: "set soft open file limit for openldap user"
+      community.general.pam_limits:
+            domain: openldap
+            limit_type: soft
+            limit_item: nofile
+            value: "{{ perun_ldap_max_open_hard_files_user_ldap_value }}"
+      register: soft_limit
+
+    - name: "set hard open file limit for openldap user"
+      community.general.pam_limits:
+            domain: openldap
+            limit_type: hard
+            limit_item: nofile
+            value: "{{ perun_ldap_max_open_hard_files_user_ldap_value }}"
+      register: hard_limit
+
+    - name: "set system-wide open files limit"
+      ansible.posix.sysctl:
+            name: fs.file-max
+            value: "{{ perun_ldap_max_open_system_files_value }}"
+      register: fs_file_max
+
+    - name: "create systemd override config directory if it does not exist"
+      when: perun_ldap_max_open_files_limits
```

Comment:
You are already performing all tasks as a block only when this is true. You can remove "when" condition here.